### PR TITLE
chore: amend seed error on CLI tests

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #5943 

### Description:

Changes:
- combine cli test github actions into 1
- add test for dbt 1.4
- run in parallel
- use different schemas for each job to avoid seed error

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
       fail-fast: false
       matrix:
         containers: [ 1, 2, 3, 4, 5 ]
-    name: App (${{ strategy.job-index }}/${{ strategy.job-total }})
+    name: App (${{ matrix.containers }}/${{ strategy.job-total }})
```
Comment (ZeRego) [inline .github/workflows/e2e-tests.yml:32]: Fix job names since `strategy.job-index` starts at 0

File: .github/workflows/cli-tests.yml [hunk 2]
```diff
@@ -69,8 +76,10 @@ jobs:
           browser: chrome
           project: ./packages/e2e
           spec: packages/e2e/cypress/cli/**/*
-          config: 'baseUrl=${{steps.deployment_url.outputs.url}}'
+          config: 'baseUrl=${{needs.prepare-preview.outputs.url}}'
         env:
           CYPRESS_PGHOST: ${{secrets.PGHOST}}
           CYPRESS_PGPASSWORD: ${{secrets.PGPASSWORD}}
+          SEED_SCHEMA: jaffle-${{strategy.job-index}}
```
Comment (rephus) [inline .github/workflows/cli-tests.yml:83]: I think this should be CYPRESS_SEED_SCHEMA if you want to get this using `Cypress.env('SEED_SCHEMA')`

File: .github/workflows/cli-tests.yml [hunk 3]
```diff
@@ -18,23 +20,28 @@ jobs:
...
+        id: deployment-url
+        run: echo "url=https://lightdash-pr-${{steps.regex.outputs.first_match}}.onrender.com" >> "$GITHUB_OUTPUT"
+
+  cli-tests:
+    runs-on: ubuntu-latest
+    needs: prepare-preview
+    strategy:
+      fail-fast: false
+      matrix:
+        dbt: ['1.3', '1.4', '1.5' ]
```
Comment (rephus) [inline .github/workflows/cli-tests.yml:32]: :kissing: :ok_hand:

**Review by rephus — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e-tests.yml [hunk 4]
```diff
@@ -93,7 +93,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        timezone: ['UTC', 'America/New_York', 'Europe/Madrid', 'Asia/Tokyo' ]
+        timezone: ['America/New_York', 'Europe/Madrid', 'Asia/Tokyo' ]
```
Comment (ZeRego) [inline .github/workflows/e2e-tests.yml:96]: removed UTC since this is already covered in the App tests

File: .github/workflows/e2e-tests.yml [hunk 5]
```diff
@@ -78,7 +78,7 @@ jobs:
       - uses: actions/upload-artifact@v3
         if: failure()
         with:
-          name: cypress-screenshots-and-videos-{{ strategy.job-index }}
+          name: cypress-screenshots-and-videos-${{ strategy.job-index }}
```
Comment (ZeRego) [inline .github/workflows/e2e-tests.yml:81]: fix typo

File: .github/workflows/e2e-tests.yml [hunk 6]
```diff
@@ -28,8 +28,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        containers: [ 1, 2, 3, 4, 5 ]
-    name: App (${{ strategy.job-index }}/${{ strategy.job-total }})
+        containers: [ 1, 2, 3 ]
```
Comment (ZeRego) [inline .github/workflows/e2e-tests.yml:31]: reduced amount of jobs since the PR env was struggling. we can try to increase it again later.

**Review by rephus — APPROVED** [review]

**Pull request merged**